extensions/positron-python/src/client/positron/manager.ts

```diff
@@ -101,8 +102,41 @@ export class PythonRuntimeManager implements IPythonRuntimeManager {
      * Recommend a Python language runtime based on the workspace.
      */
     async recommendedWorkspaceRuntime(): Promise<positron.LanguageRuntimeMetadata | undefined> {
-        // TODO: This is where we could recommend a runtime based on the
-        // workspace, e.g. if it contains a virtualenv
+        // TODO: may need other handling for multiroot workspaces
+        const workspaceUri = vscode.workspace.workspaceFolders?.[0]?.uri;
+        const userInterpreterSettings = getUserDefaultInterpreter(workspaceUri);
+        let interpreterPath: string | undefined;
+        let isImmediate = false;
+
+        if (!workspaceUri) {
+            if (userInterpreterSettings.globalValue) {
+                interpreterPath = userInterpreterSettings.globalValue;
+            } else {
+                traceInfo('No recommended runtime for workspace.');
+                return undefined;
+            }
+        } else if (await hasFiles(['.venv/**/*'])) {
```

Comment:
Is this paradigm used elsewhere in the codebase? I worry that if the venv directory is [named something else](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#create-a-new-virtual-environment) it won't get picked up. But if we search for the presence of `.venv/` elsewhere in the code then I'm comfortable saying it's outside the scope of this PR to support venvs named something else.

Reply:
Maybe we could just check for `*/bin/python` and use whatever that resolves to?

Reply:
We look for a directory specifically named `.venv` in discovery, to see if a workspace should activate a Python runtime, but I like the idea of checking for `*/bin/python`. Maybe we do that first, and if there are multiple matches, we can decide priority.


https://github.com/posit-dev/positron/blob/f6e404bf4a8900501827f8fff836fae6712c631e/extensions/positron-python/src/client/positron/discoverer.ts#L71